UnityProject/Assets/Scripts/Items/Renameable.cs

```diff
@@ -0,0 +1,119 @@
+using Mirror;
+using UnityEngine;
+using WebSocketSharp;
+
+public class Renameable : NBHandActivateInteractable, IRightClickable
+{
+	public NetTabType NetTabType = NetTabType.Rename;
+
+	[SyncVar(hook = nameof(SyncOriginalName))]
+	public string originalName;
+
+	[SyncVar(hook = nameof(SyncCustomName))]
+	public string customName;
```

Comment:
I don't quite understand this – why make these two syncvars?
from what I understand, `ItemAttributes.SetItemName()` is only supposed to be called on server, why share contents of these fields with all clients? server can do it on its own

Reply:
So it looks like these might be needed for late joiners renaming an object- they need to be able to know the original name in case they remove the rename. Setting the custom name comes from a [Command] which I think is client side?

Reply:
Clients don't have to know original name. You can make server interpret empty string in Cmd as a request to restore original name and set syncvar from that field on server